services/brig/src/Brig/User/Auth.hs

```diff
@@ -222,4 +228,32 @@ ssoLogin (SsoLogin uid label) typ = do
             AuthSuspended          -> throwE LoginSuspended
             AuthEphemeral          -> throwE LoginEphemeral
             AuthInvalidUser        -> throwE LoginFailed
-    newAccess uid typ label
+    newAccess @ZAuth.User @ZAuth.Access uid typ label
+
+-- | Log in as a LegalHold service, getting LegalHoldUser/Access Tokens.
+legalHoldLogin :: LegalHoldLogin -> CookieType -> ExceptT LegalHoldLoginError AppIO (Access ZAuth.LegalHoldUser)
+legalHoldLogin (LegalHoldLogin uid label) typ = do
+    Data.reauthenticate uid Nothing `catchE` \case
+        ReAuthMissingPassword -> pure ()
```

Comment:
This looks very wrong to me.  If it's not, it could probably use a comment.  I would have expected something like this:

```suggestion
        ReAuthMissingPassword -> throwE e
```

Reply:
Good point, thanks, I forgot about this. Galley will then have to pass on the password to brig.